cpp/doc/Doxyfile

```diff
@@ -892,7 +892,7 @@ EXCLUDE_SYMLINKS       = NO
 # Note that the wildcards are matched against the file with absolute path, so to
 # exclude all test directories for example use the pattern */test/*
 
-EXCLUDE_PATTERNS       =
+EXCLUDE_PATTERNS       = *.cpp
```

Comment:
This should not be needed. If there is a specific region of one cpp file that is causing a problem, it can be wrapped in `\\@cond` - `\\@endcond`.

Reply:
I'm not so keen on changing the DOLFIN source code to work around bugs in old versions of doxygen. I'd rather localise work-arounds in the doxygen files.

Reply:
But this is not local: `*.cpp`